GameDataParser/Parsers/SkillParser.cs

```diff
@@ -13,101 +14,127 @@ namespace GameDataParser.Parsers
+                                }
+                            }
+                            skillLevel.Add(new SkillLevel(level, spirit, upgradeLevel, upgradeSkillId, upgradeSkillLevel));
+                            metadata.SkillLevel = skillLevel[0];
                         }
-                        if (motion.Attributes.GetNamedItem("motionEffect") != null)
+                    }
+                    skillList.Add(metadata);
+                }
+                // Parsing SubSkills
+                else if (xmlFile.Name.StartsWith("table/job"))
+                {
+                    XmlDocument document = m2dFile.GetDocument(xmlFile.FileHeader);
+                    XmlNodeList job = document.SelectNodes("/ms2/job");
+                    foreach (XmlNode node in job)
+                    {
+                        if (node.Attributes.GetNamedItem("feature") != null)
                         {
-                            motionEffect = motion.Attributes["motionEffect"].Value;
+                            string feature = node.Attributes["feature"].Value;
+                            if (feature == "JobChange_02")
+                            {
+                                int jobCode = int.Parse(node.Attributes.GetNamedItem("code").Value);
+                                XmlNode skills = node.SelectSingleNode("skills");
+                                for (int i = 0; i < skills.ChildNodes.Count; i++)
+                                {
+                                    int id = int.Parse(skills.ChildNodes[i].Attributes["main"].Value);
+                                    int[] sub = new int[0];
+                                    SkillMetadata skill = skillList.Find(x => x.SkillId == id);
+                                    skill.Job = jobCode;
+                                    if (skills.ChildNodes[i].Attributes["sub"] != null)
+                                    {
+                                        if (skillList.Select(x => x.SkillId).Contains(id))
+                                        {
+                                            sub = Array.ConvertAll(skills.ChildNodes[i].Attributes["sub"].Value.Split(","), int.Parse);
+                                            skill.SubSkill = sub;
+                                            for (int n = 0; n < sub.Length; n++)
+                                            {
+                                                if (skillList.Select(x => x.SkillId).Contains(sub[n]))
+                                                {
+                                                    skillList.Find(x => x.SkillId == sub[n]).Job = jobCode;
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                            }
```

Comment:
What in the world is this doing? The sub skills just need to be an array of skill ids

Reply:
This is getting all the Skill that contains a subSkill and adding in into the corresponding skill, and because is from another xml in the same file is getting filtered under `feature=JobChange_02` that contains all the list, same as it was doing with `string/skill_{}` before.

Didn't get a better solution so far.

Reply:
The skills in table/job have an attribute with the corresponding sub skills, we can just pull the ids from there and put them into an array.

Reply:
`sub = Array.ConvertAll(skills.ChildNodes[i].Attributes["sub"].Value.Split(","), int.Parse);`
That is what im doing there, then im adding on each skill the sub it correspond with `skill.SubSkill = sub;`
And because they also are skills from the same `code` i need to go through each subskill and add them the jobCode as well.

Reply:
Yeah but I'm saying we don't need to have anything other than the id, once you have the id and we want to level those skills up we can just go through the subskill array, the sub skills should all already be in the skill array, they should all already be defined, unless they aren't?

Reply:
They aren't define on `skills` they are define either on `table/job` or `string/skills{number}` and `table/job` has them all and the default ones, we just grab the subskills and the defaults skill from that and added into the list we already have.

Reply:
And this is doing the check of the id for each job and adding the `jobCode` to that skill, and then is adding the `sub skill array` to the skill corresponding, and then on each `subskill` is adding the `jobCode` as well